chcemvediet/apps/inforequests/views/attachment.py

```diff
@@ -62,11 +63,16 @@ def attachment_download(request, attachment_pk):
 
 @require_http_methods([u'HEAD', u'GET'])
 def attachment_finalization_download(request, attachment_finalization_pk):
-    attachment_finalization = AttachmentFinalization.objects.get_or_404(
+    attachment_finalization = AttachmentFinalization.objects.successful().get_or_404(
             pk=attachment_finalization_pk)
     generic_object = attachment_finalization.attachment.generic_object
     if isinstance(generic_object, Action):
         if generic_object.branch.inforequest.published:
+            try:
+                if attachment_finalization.content is None:
```

Comment:
@martinmacko47 Property `content` vlastne robi presne to iste ako moja funkcia `attachment_file_is_working`, akurat property obsah suboru aj precita a zaroven aj zaloguje error, ale iba v pripade, ak `file.name` ma hodnotu a subor neexistuje. Mozem si pomoct s touto property?

Reply:
@viliambalaz Property `content` by som nepouzival, pretoze v `send_file_response` obsah suboru necitame z `attachment.content`, ale rucne. Cize z rovnakeho dovodu ako pri `attachment_file_is_working` by som toto nerobil. Nie je dobre nieco kotrolovat inac, nez sa to potom pouziva. Pretoze v jenotlivych implementaciach nacitavania suboru mozu byt rozdiely.

Spravme kontrolu iba na to, ze je definovane `file.name`. Ale iba v `DownloadAdminMixin.download_view`. Pretoze v `attachment_finalization_download` ma platit predsa invariant, ze ak je finalization successful, tak `file.name` ma mat. Ak ho nema, tak je to programatorska chyba a treba vratit 500 Internal server error. Aby nas to pinglo a chybu opravili.

V `DownloadAdminMixin.download_view` naopak takyto invarianet nemame. A `file.name` moze byt nedefinovane. V takom pripade treba vratit 404. Tj. ze subor pre danu instancu neexistuje.

Reply:
:ok: